### ISSUE TITLE
Question about training memory

### ISSUE THREAD

**Issue by GuangyuWang99** [ewrfcas/Uni3C#3]
Hi, 

Thanks for the great work! 

I am trying to reproduce the camera controlled version of this work, following the details described in Sec.4.1 and 5.1 of the arxiv paper. Specifically, the controlnet is added to the first 20 layers of the base model (WanI2V-14B, 480 × 768 × 81, BF16) with no cross-attention modules and 1024 hidden dim. However, this setup for training raises an OOM error on Nvidia A100 80G GPUs. 

After carefully tuning the configs, I found 80G memory can at most support the training of a 18 layer controlnet with 64 hidden dim, which contains ~10x less learnable parameters than the 0.95B controlnet described in the paper. 

Therefore, I am wondering if you use pipeline parallelism during training? Given the description from your paper: _We train the model for 6,000 steps with a batch size of **32**. The training is accomplished with **64** H100 GPUs for 40 hours._

Thanks in advance and looking forward to your reply.

**Comment by ewrfcas** [comment]
Thanks for your attention. There would be some additional efficient strategies to improve the training, including FSDP and sequence parallel. Moreover, our inference codes and models would be released before the end of this month.

**Comment by GuangyuWang99** [comment]
Thanks for your prompt reply. The open source of such a model will be of great contribution to the community:)

**Comment by MinusOne2** [comment]
> Thanks for your attention. There would be some additional efficient strategies to improve the training, including FSDP and sequence parallel. Moreover, our inference codes and models would be released before the end of this month.

Hi, I want to ask the batch size of 32 means total batch size or sigle batch size per GPU?

**Comment by wangyichen191** [comment]
> Hi,
> 
> Thanks for the great work!
> 
> I am trying to reproduce the camera controlled version of this work, following the details described in Sec.4.1 and 5.1 of the arxiv paper. Specifically, the controlnet is added to the first 20 layers of the base model (WanI2V-14B, 480 × 768 × 81, BF16) with no cross-attention modules and 1024 hidden dim. However, this setup for training raises an OOM error on Nvidia A100 80G GPUs.
> 
> After carefully tuning the configs, I found 80G memory can at most support the training of a 18 layer controlnet with 64 hidden dim, which contains ~10x less learnable parameters than the 0.95B controlnet described in the paper.
> 
> Therefore, I am wondering if you use pipeline parallelism during training? Given the description from your paper: _We train the model for 6,000 steps with a batch size of **32**. The training is accomplished with **64** H100 GPUs for 40 hours._
> 
> Thanks in advance and looking forward to your reply.

Hello, have you reproduced the training code? If so, could you please provide it? Thank you very much.
